VerilogForVerificationStyle.md

```diff
@@ -0,0 +1,850 @@
+  * [Bind Statements](#bind-statements)
+  * [Simulator Specific Code](#simulator-specific-code)
+  * [Forbidden System Tasks and Functions](#forbidden-system-tasks-and-functions)
+
+
+
+## Introduction
+
+This guide defines the Comportable style for DV-specific SystemVerilog code. The
+goal is to use this common set of guidelines when implementing testbenches using
+the UVM methodology to allow all developed Comportable DV code to be uniform,
+reusable, and portable.
+This document assumes previous experience and working knowledge of the UVM, and
+as such all included code snippets will be fairly concise.
+A testbench generation script
+[uvmdvgen.py](https://github.com/lowRISC/opentitan/blob/master/util/uvmdvgen.py)
+is used to automatically generate boilerplate DV testbench classes that follow
+these style guidelines.
+
+## Naming and Style
+
+Refer to [lowRISC Verilog naming
+conventions](https://github.com/lowRISC/style-guides/blob/master/VerilogCodingStyle.md#naming)
+for any general naming style guidelines. In general, common UVM testbench
+components will be declared and assigned handles as follows:
+
+| Testbench component               | Style                                        |
+| --------------------------------- | -------------------------------------------- |
+| Virtual interfaces                | `virtual <if_name>_if <if_name>_vif;`        |
+| Environment                       | `<dut>_env env;`                             |
+| Env level config object           | `rand <dut>_env_cfg cfg;`                    |
+| Env coverage collection object    | `<dut>_env_cov cov;`                         |
+| Agent                             | `<dut>_agent m_<dut>_agent;`                 |
+| Agent level config object         | `rand <dut>_agent_cfg m_<dut>_agent_cfg;`    |
+| Agent coverage collection object  | `<dut>_agent_cov cov;`                       |
+| Driver                            | `<dut>_driver driver;`                       |
+| Monitor                           | `<dut>_monitor monitor;`                     |
+| Scoreboard                        | `<dut>_scoreboard scoreboard;`               |
+| Virtual sequencer                 | `<dut>_virtual_sequencer virtual_sequencer;` |
+| Sequencer                         | `<dut>_sequencer <dut>_sequencer;`           |
+
+The UVM testbench generation script will automatically create skeletons of these
+components with this naming scheme, so almost all of this naming is automated
+out.
+Additional naming guidelines:
+
+1. The top level testbench file that instantiates the DUT must be named `tb.sv`.
+   The testbench module must be named `tb`, and the instance name of the DUT in
+   `tb.sv` must be `dut`. Using this standardized naming allows universal usage
+   of any supporting DV sources (coverage cfg files, and so on).
+2. When instantiating any objects, the `name` argument to `type_id::create()`
+must match the name of the variable being assigned to.
```

Comment:
nit: alignment of following lines

Reply:
Thanks for the pointer, I've fixed this alignment of number/bullet lists throughout the document.